IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m

```diff
@@ -81,21 +84,39 @@ - (BOOL)updateAuthorityURL:(NSURL *)cacheAuthorityURL
     if (!cacheAuthorityURL
         || [cacheAuthorityURL isEqual:requestAuthorityURL]) return YES;
     
-    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWitHomeAccountId:homeAccountId clientId:clientId];
-    MSIDAccountMetadataCacheItem *accountMetadataItem = [_metadataCache accountMetadataWithKey:key context:context error:error];
-    if (!accountMetadataItem)
+    NSError *localError;
+    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWithClientId:clientId];
+    MSIDAccountMetadataCacheItem *cacheItem = [_metadataCache accountMetadataCacheItemWithKey:key context:context error:&localError];
+    if (localError)
+    {
+        if (error) *error = localError;
+        return NO;
+    }
+    
+    if (!cacheItem)
+    {
+        cacheItem = [[MSIDAccountMetadataCacheItem alloc] initWithClientId:clientId];
+    }
+    
+    MSIDAccountMetadata *accountMetadata = [cacheItem accountMetadataForHomeAccountId:homeAccountId];
+    if (!accountMetadata)
+    {
+        accountMetadata = [[MSIDAccountMetadata alloc] initWithHomeAccountId:homeAccountId clientId:clientId];
+    }
+    
+    if (![accountMetadata setCachedURL:cacheAuthorityURL forRequestURL:requestAuthorityURL instanceAware:instanceAware error:error])
```

Comment:
will this also check if existing authority URL is the same as new one, and not do writing if not needed?

Reply:
Later `MSIDMetadataCache` will check whether the item to store is equal to the existing one.
But I agree with you that we could improve more by adding a checking here as well. 

Changes made :)